server/api/src/rsDataObjRename.cpp

```diff
@@ -129,20 +132,46 @@ namespace
         }
         else {
             if ( ( status = isData( rsComm, srcDataObjInp->objPath, &srcId ) ) >= 0 ) {
-                if ( isData( rsComm, destDataObjInp->objPath, &destId ) >= 0 &&
-                        getValByKey( &srcDataObjInp->condInput, FORCE_FLAG_KW ) != NULL ) {
-                    /* dest exist */
-                    rsDataObjUnlink( rsComm, destDataObjInp );
-                }
                 srcDataObjInp->oprType = destDataObjInp->oprType = RENAME_DATA_OBJ;
                 status = getDataObjInfo( rsComm, srcDataObjInp, &dataObjInfoHead, ACCESS_DELETE_OBJECT, 0 );
-
                 if ( status < 0 ) {
                     rodsLog( LOG_ERROR,
                             "%s: src data %s does not exist, status = %d",
                             __FUNCTION__, srcDataObjInp->objPath, status );
                     return status;
                 }
+
+                if (const auto ret = ill::try_lock(*dataObjInfoHead, ill::lock_type::write); ret < 0) {
+                    const auto msg = fmt::format(
+                        "rename not allowed because source data object is locked "
+                        "[error code=[{}], source path=[{}]",
+                        ret, srcDataObjInp->objPath);
+
+                    //addRErrorMsg(&rsComm->rError, ret, msg.data());
```

Comment:
dead?

Reply:
I couldn't decide if there should be a message back to the user or just let the error code do the talking. There's also the matter of `imv -r` spewing a lot of messages, etc.

Reply:
right, certainly don't want recursive being noisy.    let's leave it out for now - and let the error code do the work for now.   can always make it more helpful/friendly later.

Reply:
Yep, I'll take it out, then